plugins/network/smtp_hello_

```diff
@@ -20,37 +17,227 @@
+
+    [smtp_hello_*]
+    env.SMTP_PORTS 25 465 587
+    env.SMTP_COMMAND HELO localhost
+    env.TIMEOUT_SECONDS 60
+
+=back
+
+Default warnings and criticals are as following:
+Emit a warning when we take more than half the configured timeout time (60s). Set to critical when we get more than 99%
+ of the timeout time (118s). Those are calculated as an integer, for timeouts with lower numbers also configure individual
+ warnings and criticals. The plugin also emits a warning when curl is not able to communicate properly with the SMTP
+ server.
+
+Different warnings and critical can be specified as following:
+
+=over 2
+
+    [smtp_hello_*]
+    # warn when any port doesn't respond within 20 seconds (note make sure to overwrite this for status)
+    env.warning :10
+    # warn when port 25 doesn't respond within 20 seconds
+    env.host_warning :20
+    # critical when port 25 doesn't respond within 60 seconds
+    env.host_critical :60
+    # any status code that is not zero should be a warning
+    env.status_warning 0:0
+    # warn when port 25 doesn't respond within 40 seconds
+    env.host_465_warning :40
+    # critical when port 465 doesn't respond within 60 seconds
+    env.host_465_critical :60
+    # any status code that is not zero should be a warning
+    env.status_465_warning 0:0
+
+=back
+
+
+ [postfix_mailqueue]
+    env.spooldir /var/spool/postfix
+
+=head1 AUTHOR
+
+Marek Mahut
+
+Copyright (C) 2008 Marek Mahut <mmahut@fedoraproject.org>
+Copyright (C) 2022 Andreas Perhab, WT-IO-IT GmbH <a.perhab@wtioit.at>
+
+=head1 LICENSE
+
+Gnu GPLv2
+
+SPDX-License-Identifier: GPL-2.0-or-later
 
-        response=`echo HELO localhost | /usr/bin/time -f "%e" nc -C -w 120 $host 25 2>&1 | tail -n 1`
-        echo "host.value $response"
 
+
+=head1 MAGIC MARKERS
+
+=begin comment
+
+These magic markers are used by munin-node-configure when installing
+munin-node.
+
+=end comment
+
+ #%# family=auto
+ #%# capabilities=autoconf
+
+=cut
+
+. "$MUNIN_LIBDIR/plugins/plugin.sh"
+
+if [ "${MUNIN_DEBUG:-0}" == "1" ]; then
+  set -x
+fi
+
+SMTP_COMMAND=${SMTP_COMMAND:-HELO localhost}
+SMTP_PORTS=${SMTP_PORTS:-25}
+TIMEOUT_SECONDS=${TIMEOUT_SECONDS:-120}
+
+host=$(basename "$0" | sed 's/^smtp_hello_//g')
+
+time_executable=""
+if which time >/dev/null; then
+  # use time executable found in PATH
+  time_executable=$(which time)
+elif [ -x /usr/bin/time ]; then
+  # use time executable found in /usr/bin
+  time_executable="/usr/bin/time"
+fi
+if [ "$time_executable" != "" ]; then
+  function take_the_time() {
+    "$time_executable" -f "%e" "$@" 2>&1 | tail -n 1
+    return "${PIPESTATUS[0]}"
+  }
+elif (time echo) >/dev/null 2>&1; then
+  # if we cannot find time executable but time keyword is successful we use the keyword
+  function take_the_time() {
+    # we use LC_ALL=C to force . as decimal separator on systems where LC_ is set to a language that uses ,
+    (
+      TIMEFORMAT='%R'
+      LC_ALL=C
+      time "$@" 1>/dev/null
+    ) 2>&1
```

Comment:
I think, you can reduce this down to the following:
```shell
{ TIMEFORMAT='%R' LC_ALL=C time "$@" 1>/dev/null; } 2>&1
```

At least for me it is easier to read (no complications with `$?`, no sub-shell).
But this may be a matter of taste.

Reply:
Ok i need to read up on the `{...;}`, this works as well to achive the stderr redirection mentioned above.

But: (sorry)
On systems without the time executable (usually `/usr/bin/time`)
```bash
{ TIMEFORMAT='%R' LC_ALL=C time echo "OK" 1>/dev/null; } 2>&1
```
gives an error:
```
-bash: time: command not found
```
whereas on systems with the time executable it clearly shows the output from the time command (ignoring the variables for the time keyword):
```
0.00user 0.00system 0:00.00elapsed ?%CPU (0avgtext+0avgdata 1452maxresident)k
0inputs+0outputs (0major+67minor)pagefaults 0swaps
```

eventually you missed the intentional newlines:
```bash
TIMEFORMAT='%R'
LC_ALL=C
time "$@" 1>/dev/null
```
is not equivalent to
```bash
TIMEFORMAT='%R' LC_ALL=C time "$@" 1>/dev/null
```
but should be
```bash
TIMEFORMAT='%R'; LC_ALL=C; time "$@" 1>/dev/null
```
which would work:
```bash
{ TIMEFORMAT='%R'; LC_ALL=C; time echo "OK" 1>/dev/null; } 2>&1
```
gives
```
0.000
```
this then would set `TIMEFORMAT` and `LC_ALL` for the following script when used with `{...;}` instead of `(...)`, so i think the subshell is ok here (even if i do not know of any sideeffects those 2 variables would have for the rest of the script, i would like to make sure they are contained in the subshell)